crates/red_knot_python_semantic/src/types/subclass_of.rs

```diff
@@ -68,6 +68,15 @@ impl<'db> SubclassOfType<'db> {
         Type::from(self.subclass_of).member(db, name)
     }
 
+    /// A class `T` is an instance of its metaclass `U`,
+    /// so the type `type[T]` is a subtype of the instance type `U`.
+    pub(crate) fn as_instance_type_of_metaclass(&self, db: &'db dyn Db) -> Type<'db> {
+        match self.subclass_of {
+            ClassBase::Dynamic(_) => KnownClass::Type.to_instance(db),
```

Comment:
I think this line is OK as-is, as far as this method is used in this PR, because our semantics of non-fully-static types in `is_disjoint_from` is that we treat them as equivalent to their widest possible type (that is, `Any` and `object` are treated the same by `is_disjoint_from`, in being disjoint from nothing.) But if this method were to be used in other contexts in future, we need to be a little careful about this line, it may not be the right thing; we should maybe return `Dynamic` instead.

Reply:
Hmm, good point. In my first version of this PR I used this method from multiple places (which was why I created at method)... but it looks like now it's only used in a single location. So maybe I should just inline it at that single location (like it was prior to this PR)?

Reply:
Yeah, that might be better, in that it would reduce the risk of reusing this logic where this isn't the right handling of it.